File: S3-2.scala
val data = spark.sparkContext.parallelize(Array(('A',1),('b',2),('c',3)))
val data2 =spark.sparkContext.parallelize(Array(('A',4),('A',6),('b',7),('c',3),('c',8)))
val result = data.join(data2)
println(result.collect().mkString(","))


val rdd1 = sc.parallelize(Seq((1,"jan",2016),(3,"nov",2014, (16,"feb",2014)))
val rdd2 = sc.parallelize(Seq((5,"dec",2014),(1,"jan",2016)))
val comman = rdd1.intersection(rdd2)
comman.foreach(Println)
comman.collect()

val rdd1 = sc.parallelize(Seq((1,"jan",2016),(3,"nov",2014),(16,"feb",2014),(3,"nov",2014)))
val result = rdd1.distinct()
println(result.collect().mkString(", "))


val data = sc.parallelize(Array(('k',5),('s',3),('s',4),('p',7),('p',5),('t',8),('k',6)),3)
val group = data.groupByKey().collect()
group.foreach(println)


val words = Array("one","two","two","four","five","six","six","eight","nine","ten")
val data = sc.parallelize(words).map(w => (w,1)).reduceByKey(_+_)
data.foreach(println)


val data = sc.parallelize(Seq(("maths",52), ("english",75), ("science",82), ("computer",65), ("maths",85)))
val sorted = data.sortByKey()
sorted.foreach(println)


val rdd1 = sc.parallelize(Array("jan","feb","mar","april","may","jun"),3)
val result = rdd1.coalesce(2)
result.foreach(println)

File: S3-2.py
data = sc.parallelize([('A',1),('b',2),('c',3)])
data2 =sc.parallelize([('A',4),('A',6),('b',7),('c',3),('c',8)])
result = data.join(data2)
print(result.collect())


rdd1 = sc.parallelize([(1,"jan",2016),(3,"nov",2014, (16,"feb",2014))])
rdd2 = sc.parallelize([(5,"dec",2014),(1,"jan",2016)])
comman = rdd1.intersection(rdd2)
comman.collect()


rdd1 = sc.parallelize([(1,"jan",2016),(3,"nov",2014),(16,"feb",2014),(3,"nov",2014)])
result = rdd1.distinct()
print(result.collect())


data = sc.parallelize([('k',5),('s',3),('s',4),('p',7),('p',5),('t',8),('k',6)],3)
group = data.groupByKey().collect()
for x in group:\
	print(x)

words = ["one","two","two","four","five","six","six","eight","nine","ten"]
data = sc.parallelize(words).map(lambda w : (w,1))
data2 = data.reduceByKey(lambda accum, n: accum + n)
data2.collect()


data = sc.parallelize([("maths",52), ("english",75), ("science",82), ("computer",65), ("maths",85)])
sorted = data.sortByKey()
sorted.collect()


rdd1 = sc.parallelize(["jan","feb","mar","april","may","jun"],3)
result = rdd1.coalesce(2)
result.collect()


def f(x): print(x)   

data = sc.parallelize([('k',5),('s',3),('s',4),('p',7),('p',5),('t',8),('k',6)],3)
group = data.groupByKey()
group.foreach(f)



